zkevm-circuits/src/keccak_circuit/keccak_padding_table.rs

```diff
@@ -0,0 +1,650 @@
+
+                Ok(())
+            },
+        )
+    }
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+//////////////////////////////////////////////////////////////////////////////////////////////////////////// 
+
+#[derive(Clone, Debug)]
+#[allow(missing_docs)]
+pub struct KeccakPaddingConfig<F> {
+    q_enable: Selector,
+    q_end: Column<Advice>,
+    d_bytes: [Column<Advice>; KECCAK_WIDTH_IN_BYTES],
+    d_lens: [Column<Advice>; KECCAK_RATE_IN_BYTES],
+    d_rlcs: [Column<Advice>; KECCAK_RATE_IN_BYTES],
+    is_pads: [Column<Advice>; KECCAK_RATE_IN_BYTES],
+    is_pad_starts: [Column<Advice>; KECCAK_RATE_IN_BYTES],
+    is_pad_ends: [Column<Advice>; KECCAK_RATE_IN_BYTES],
+    randomness: Column<Advice>,
+    _marker: PhantomData<F>,
+    padding_combinations_table: PaddingCombinationsConfig<F>,
+}
+
+impl<F: Field> KeccakPaddingConfig<F> {
+    pub(crate) fn configure(meta: &mut ConstraintSystem<F>) -> Self {
+        let randomness = meta.advice_column();        
+        let q_enable = meta.selector();
+        let q_end = meta.advice_column();
+
+        let padding_combinations_table = PaddingCombinationsConfig::<F>::configure(meta);
+
+        let d_bytes = [(); KECCAK_WIDTH_IN_BYTES].map(|_| meta.advice_column());
+        let d_lens = [(); KECCAK_RATE_IN_BYTES].map(|_| meta.advice_column());
+        let d_rlcs = [(); KECCAK_RATE_IN_BYTES].map(|_| meta.advice_column());
+        let is_pads = [(); KECCAK_RATE_IN_BYTES].map(|_| meta.advice_column());
+        let is_pad_starts = [(); KECCAK_RATE_IN_BYTES].map(|_| meta.advice_column());
+        let is_pad_ends = [(); KECCAK_RATE_IN_BYTES].map(|_| meta.advice_column());
+        
+        // Check that (padding flag, data/padding byte) combinations are restricted to the combinations in padding_combinations_table.        
+        // The table will estrict padding flags to {0,1} and data/padding bytes to [0, 255].
+        // Only particular combinations are allowed, check PaddingCombinationsConfig load method comments.
+        for i in 0..is_pads.len() {
+            meta.lookup("Check allowed data/padding/flag combinations", |meta| {
+                // let q_enable = meta.query_selector(q_enable);
+                vec![
+                    (meta.query_advice(d_bytes[i], Rotation::cur()), padding_combinations_table.byte_col),
+                    (meta.query_advice(is_pads[i], Rotation::cur()), padding_combinations_table.is_padding_col),
+                    (meta.query_advice(is_pad_starts[i], Rotation::cur()), padding_combinations_table.is_padding_start_col),
+                    (meta.query_advice(is_pad_ends[i], Rotation::cur()), padding_combinations_table.is_padding_end_col),
```

Comment:
I know you want to try a new method, but the "normal" way (like the byte version) would be to use the `s_padding_step` and `is_pad_last` expressions (as created in the byte version) as inputs here in the table. So you would not create new columns here to put that data to lower the amount of columns needed (unless you need to do this to lower the expression degree to a certain level).